src/implementation.ts

```diff
@@ -107,14 +107,15 @@ function batchLoaderLogic(resourceConfig: BatchResourceConfig, resourcePath: Rea
                     return `
                         // For now, we assume that the dataloader key should be the first argument to the resource
                         // @see https://github.com/Yelp/dataloader-codegen/issues/56
-                        const resourceArgs = [{
+                        const resourceArg = {
                             ..._.omit(requests[0], '${resourceConfig.newKey}'),
                             ${batchKeyParam},
-                        }];
+                        };
                     `;
                 })()}
 
-                let response = await ${callResource(resourceConfig, resourcePath)}(resourceArgs);
+                // Any-type so this is re-assignable to the 'nestedPath' without TS complaining
```

Comment:
in theory this could use a supression instead

but then we don't necessarily know what linter or whatever we'll use in the future that might complain, which might have a different suppression syntax (e.g. maybe one day we switch to biome)

so i guess this is still good